internal/dag/builder.go

```diff
@@ -512,6 +512,91 @@ func (b *Builder) computeHTTPProxy(proxy *projcontour.HTTPProxy) {
 	}
 }
 
+// expandPrefixMatches adds new Routes to account for the difference
+// between prefix replacement when matching on '/foo' and '/foo/'.
+//
+// The table below shows the behavior of Envoy prefix rewrite. If we
+// match on only `/foo` or `/foo/`, then the unwanted rewrites marked
+// with X can result. This means that we need to generate separate
+// prefix matches (and replacements) for the two cases.
+//
+// | Matching Prefix | Replacement | Client Path | Rewritten Path |
+// |-----------------|-------------|-------------|----------------|
+// | `/foo`          | `/bar`      | `/foosball` |   `/barsball`  |
+// | `/foo/`         | `/bar`      | `/foo/type` | X `/bartype`   |
+// | `/foo`          | `/bar/`     | `/foosball` | X `/bar/sball` |
+// | `/foo/`         | `/bar/`     | `/foo/type` |   `/bar/type`  |
```

Comment:
Can you please add cases for

match: /foo, replace: /, client path: /foo/v1, rewritten: /v1

Reply:
This table is demonstrating the envoy behavior, so in the example you give, the rewritten path  would be `//v1`.

Reply:
I'm sorry but we must address this problem. match: /foo, replace: /, client path: /foo/v1, must be rewritten as /v1 to resolve #899

Reply:
Yes, we do address this, see the "replaceWithSlash" test.

This table is showing "the behavior of Envoy prefix rewrite". It is not showing the behavior of how Contour configures the rewrite. In this comment, I'm trying to explain the rationale for why we need to generate new routes for the different rewrite cases.